src/components/SerialConsole/SerialConsole.js

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import XTerm from './XTerm';
+
+class SerialConsole extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.focusTerminal = this.focusTerminal.bind(this);
+    this.onResetClick = this.onResetClick.bind(this);
+    this.onDisconnectClick = this.onDisconnectClick.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.onConnect();
+  }
+
+  componentWillUnmount() {
+    this.props.onDisconnect();
+  }
+
+  onResetClick(event) {
+    if (event.button !== 0) return;
+
+    this.props.onDisconnect();
+    this.props.onConnect();
+    this.focusTerminal();
+  }
+
+  onDisconnectClick(event) {
+    if (event.button !== 0) return;
+
+    this.props.onDisconnect();
+    this.focusTerminal();
+  }
+
+  /**
+   * Backend sent data.
+   */
+  onDataReceived(data) {
+    if (this.refs.terminal) {
+      this.refs.terminal.onDataReceived(data);
+    }
+  }
+
+  /**
+   * Backend closed connection.
+   */
+  onConnectionClosed(reason) {
+    if (this.refs.terminal) {
+      this.refs.terminal.onConnectionClosed(reason);
+    }
+  }
+
+  focusTerminal() {
+    this.refs.resetButton.blur();
+    this.refs.disconnectButton.blur();
+
+    if (this.refs.terminal) this.refs.terminal.focus();
+  }
+
+  render() {
+    const { id, status } = this.props;
+    const {
+      textDisconnect = 'Disconnect',
+      textDisconnected = 'Disconnected from serial console. Click the Reconnect button.',
+      textLoading = 'Loading ...',
+      textReconnect = 'Reconnect',
```

Comment:
These need to be override-able by applications using I18N. See https://github.com/patternfly/patternfly-react/pull/162/files#diff-9694875e1ead42ed0c55964e603992b1R6 as an example on how to do this.

Reply:
That was the point here as well. Anyway, wrapped into the `messages` prop to be consistent.